### PR TITLE
Much better test coverage of AbstractAdapter

### PR DESCRIPTION
**Pull request by marc-mabe**
I don't no why all of these lines were commented out. I think it was from initial writing the component as the tests where written for one of the very very first versions (before 2.0).

I have commented all these tests in and refactored the tests to the current version.

### CODE REVIEW

File: test/Storage/Adapter/AbstractAdapterTest.php [hunk 1]
```diff
@@ -338,23 +348,6 @@ public function testInternalHasItemsCallsInternalHasItem()
         $this->assertEquals(['key1'], $rs);
     }
 
-    public function testGetMetadataCallsInternalGetMetadata()
```
Comment (marc-mabe) [inline test/Storage/Adapter/AbstractAdapterTest.php]: This gets tested by `testGetMetadatas`

**Comment by marc-mabe** [comment]
The failed test is a real bug and will be fixed in #36

File: test/Storage/Adapter/AbstractAdapterTest.php [hunk 2]
```diff
@@ -416,6 +403,10 @@ public function simpleEventHandlingMethodDefinitions()
 
             ['decrementItem', 'internalDecrementItem', ['k', 1], true],
             ['decrementItems', 'internalDecrementItems', [['k1' => 1, 'k2' => 2]], []],
+
+            ['getCapabilities', 'internalGetCapabilities', [], new Capabilities(
```
Comment (Maks3w) [inline test/Storage/Adapter/AbstractAdapterTest.php]: I prefer create the object in a local var with a descriptive name so you can read the list of test cases quickly

**Comment by Maks3w** [comment]
I'll review this later

File: test/Storage/Adapter/AbstractAdapterTest.php (hunk 2 above)
Comment (marc-mabe) [inline test/Storage/Adapter/AbstractAdapterTest.php]: updated

**Pull request merged**